mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -874,195 +905,200 @@ public DuplicateInsertPreventionStrategy getDuplicateInsertPreventionStrategy(En
 		}
 
 		if ( queryParameters.getRowSelection().getMaxRows() != null ) {
-			cursor.limit( queryParameters.getRowSelection().getMaxRows() );
+			prepareFind.limit( queryParameters.getRowSelection().getMaxRows() );
 		}
-
-		return new MongoDBResultsCursor( cursor, entityKeyMetadata );
+		return new MongoDBResultsCursor( prepareFind.iterator(), entityKeyMetadata );
 	}
 
-	private static ClosableIterator<Tuple> doFindOne(final MongoDBQueryDescriptor query, final DBCollection collection,
+	private static ClosableIterator<Tuple> doFindOne(final MongoDBQueryDescriptor query, final MongoCollection<Document> collection,
 			final EntityKeyMetadata entityKeyMetadata) {
 
-		final DBObject theOne = collection.findOne( query.getCriteria(), query.getProjection() );
+		final Document theOne = collection.find( query.getCriteria() ).projection( query.getProjection() ).first();
 		return new SingleTupleIterator( theOne, collection, entityKeyMetadata );
 	}
 
-	private static ClosableIterator<Tuple> doFindAndModify(final MongoDBQueryDescriptor queryDesc, final DBCollection collection,
+	private static ClosableIterator<Tuple> doFindAndModify(final MongoDBQueryDescriptor queryDesc, final MongoCollection<Document> collection,
 			final EntityKeyMetadata entityKeyMetadata) {
 
-		DBObject query = (DBObject) queryDesc.getCriteria().get( "query" );
-		DBObject fields = (DBObject) queryDesc.getCriteria().get( "fields" );
-		DBObject sort = (DBObject) queryDesc.getCriteria().get( "sort" );
+		Document query = (Document) queryDesc.getCriteria().get( "query" );
+		Document fields = (Document) queryDesc.getCriteria().get( "fields" );
+		Document sort = (Document) queryDesc.getCriteria().get( "sort" );
 		Boolean remove = (Boolean) queryDesc.getCriteria().get( "remove" );
-		DBObject update = (DBObject) queryDesc.getCriteria().get( "update" );
-		Boolean nevv = (Boolean) queryDesc.getCriteria().get( "new" );
+		Document update = (Document) queryDesc.getCriteria().get( "update" );
+		Boolean returnNewDocument = (Boolean) queryDesc.getCriteria().get( "new" );
 		Boolean upsert = (Boolean) queryDesc.getCriteria().get( "upsert" );
 		Boolean bypass = (Boolean) queryDesc.getCriteria().get( "bypassDocumentValidation" );
-		DBObject o = (DBObject) queryDesc.getCriteria().get( "writeConcern" );
+		Document o = (Document) queryDesc.getCriteria().get( "writeConcern" );
 		WriteConcern wc = getWriteConcern( o );
-		final DBObject theOne = collection.findAndModify( query, fields, sort, ( remove != null ? remove : false ),
-				update, (nevv != null ? nevv : false), (upsert != null ? upsert : false), (bypass != null ? bypass : false),
+		//@TODO Analyse it! New API have methods findAndRemove and findAndUpdate
+		/*final Document theOne = collection.findAndModify( query, fields, sort, ( remove != null ? remove : false ),
+				update, (returnNewDocument != null ? returnNewDocument : false), (upsert != null ? upsert : false), (bypass != null ? bypass : false),
```

Comment:
@DavideD  A lot of thanks. I must to remove comments.  I will prepare new commit.